### PR TITLE
CB-8636 move cloud related validation into a flow step which improves…

### PR DESCRIPTION
**Pull request by doktoric**
… the request response times. Moved group validation, template validation, parameters like encryption keys, DL and ENV validation. Also moved kerberos config generation under the kerberos validation.

See detailed description in the commit message.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/StackCreatorService.java [hunk 1]
```diff
@@ -301,23 +273,12 @@ public StackV4Response createStack(User user, Workspace workspace, StackV4Reques
     }
 
     private void assignOwnerRoleOnDataHub(User user, StackV4Request stackRequest, Stack newStack) {
+        // TODO grpc call
```
Comment (topolyai5) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/StackCreatorService.java:276]: what kind of grpc call?

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/InstanceTemplateValidator.java [hunk 2]
```diff
@@ -4,23 +4,22 @@
...
 @Component
-public class InstanceTemplateV4RequestValidator implements Validator<InstanceTemplateV4Request> {
+public class InstanceTemplateValidator implements Validator<Template> {
 
     @Override
-    public ValidationResult validate(InstanceTemplateV4Request subject) {
+    public ValidationResult validate(Template subject) {
         ValidationResultBuilder resultBuilder = ValidationResult.builder();
         if (Objects.isNull(subject)) {
             resultBuilder.error("Template request cannot be null in the instance group request.");
```
Comment (topolyai5) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/InstanceTemplateValidator.java:19]: Maybe this validation runs in the flow

File: core/src/test/java/com/sequenceiq/cloudbreak/controller/validation/stack/StackV4RequestValidatorTest.java [hunk 3]
```diff
@@ -14,25 +15,21 @@
...
-import com.sequenceiq.cloudbreak.structuredevent.CloudbreakRestRequestThreadLocalService;
 import com.sequenceiq.cloudbreak.service.blueprint.BlueprintService;
 import com.sequenceiq.cloudbreak.service.environment.EnvironmentClientService;
 import com.sequenceiq.cloudbreak.service.rdsconfig.RdsConfigService;
+import com.sequenceiq.cloudbreak.structuredevent.CloudbreakRestRequestThreadLocalService;
 import com.sequenceiq.cloudbreak.validation.ValidationResult;
-import com.sequenceiq.cloudbreak.validation.ValidationResult.State;
 
 @RunWith(MockitoJUnitRunner.class)
 public class StackV4RequestValidatorTest extends StackRequestValidatorTestBase {
```
Comment (topolyai5) [inline core/src/test/java/com/sequenceiq/cloudbreak/controller/validation/stack/StackV4RequestValidatorTest.java:35]: Please rename the test class, and check all the test names in the PR

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/validate/cloud/CloudConfigValidationActions.java [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    private CredentialConverter credentialConverter;
+
+    @Inject
+    private StackViewService stackViewService;
+
+    @Inject
+    private ParametersValidator parametersValidator;
+
+    @Inject
+    private StackValidator stackRequestValidator;
```
Comment (topolyai5) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/validate/cloud/CloudConfigValidationActions.java:76]: please rename the field

File: core/src/main/java/com/sequenceiq/cloudbreak/service/blueprint/BlueprintService.java [hunk 5]
```diff
@@ -206,7 +206,7 @@ public ScaleRecommendation getScaleRecommendation(Long workspaceId, String bluep
     }
 
     public Set<Blueprint> getAllAvailableInWorkspace(Workspace workspace) {
-        updateDefaultBlueprintCollection(workspace);
+        //updateDefaultBlueprintCollection(workspace);
```
Comment (topolyai5) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/blueprint/BlueprintService.java:209]: please remove the slash or delete the code

File: core/src/test/java/com/sequenceiq/cloudbreak/controller/StackCreatorServiceTest.java [hunk 6]
```diff
@@ -201,8 +183,8 @@ public void shouldThrowBadRequestWhenRecipeIsMissing() {
         instanceGroupV4Request.setRecipeNames(Set.of(RECIPE_NAME));
         stackRequest.setInstanceGroups(List.of(instanceGroupV4Request));
 
-        when(validationResult.getState()).thenReturn(State.VALID);
-        when(stackRequestValidator.validate(stackRequest)).thenReturn(validationResult);
+        //when(validationResult.getState()).thenReturn(State.VALID);
```
Comment (topolyai5) [inline core/src/test/java/com/sequenceiq/cloudbreak/controller/StackCreatorServiceTest.java:186]: remove comment section

**Review by topolyai5 — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/InstanceTemplateValidator.java (hunk 2 above)
Comment (doktoric) [reply core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/InstanceTemplateValidator.java:19]: there is no provider related validation

Comment (topolyai5) [reply core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/InstanceTemplateValidator.java:19]: I mean `request` in the message

File: freeipa/src/main/java/com/sequenceiq/freeipa/converter/stack/CreateFreeIpaRequestToStackConverter.java [hunk 7]
```diff
@@ -218,7 +218,7 @@ private Json getTags(CreateFreeIpaRequest source, Stack stack, String userCrn, S
             // userdefined tags comming from environment service
             return new Json(new StackTags(userDefined, new HashMap<>(), getDefaultTags(stack, userCrn, userName)));
         } catch (Exception ignored) {
-            throw new BadRequestException("Failed to convert dynamic tags.");
+            throw new BadRequestException("Failed to convert dynamic tags." + ignored.getMessage());
```
Comment (topolyai5) [inline freeipa/src/main/java/com/sequenceiq/freeipa/converter/stack/CreateFreeIpaRequestToStackConverter.java:221]: it will conflict with my change

**Review by topolyai5 — APPROVED** [review]

**Pull request merged**